airframe-http/src/main/scala/wvlet/airframe/http/RxFilter.scala

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package wvlet.airframe.http
+
+import wvlet.airframe.http.HttpMessage.{Request, Response}
+import wvlet.airframe.rx.Rx
+
+import scala.util.control.NonFatal
+
+/**
+  * [[RxContext]] is a service interface for processing request and returning `Rx[Response]`.
+  */
+trait RxContext {
+  private[http] def backend: HttpBackend[Request, Response, Rx]
```

Comment:
The current HttpBackend implementation is a bit heavy. We only need a request/response type converter and thread-local storage. It would be better to introduce a lightweight interface.